client/schema.go

```diff
@@ -0,0 +1,405 @@
+				for enumID, enumCase := range enumCases {
+					// Convert to lower case for comparison during ingestion.
+					if caseInsensitive {
+						enumCase = strings.ToLower(enumCase)
+					}
+					// all mapping should be pre created
+					enumMappings[columnID][enumCase] = enumID
+
+					if defValuePtr != nil {
+						if *defValuePtr == enumCase {
+							enumDefaultValueMappings[columnID] = enumID
+						}
+					}
+				}
+			} else {
+				cf.metricScope.Tagged(map[string]string{
+					"table":    table.Name,
+					"columnID": strconv.Itoa(columnID),
+				}).Counter("err_fetch_enum_dict").Inc(1)
+				return utils.StackError(err, "Failed to fetch enum cases for table: %s, column: %d", table.Name, columnID)
+			}
+		}
+	}
+	cf.Lock()
+	cf.enumMappings[table.Name] = enumMappings
+	cf.enumDefaultValueMappings[table.Name] = enumDefaultValueMappings
+	cf.Unlock()
+	return nil
+}
+
+func (cf *CachedSchemaHandler) setTable(table *metaCom.Table) *TableSchema {
+	columnDict := make(map[string]int)
+	for columnID, column := range table.Columns {
+		if !column.Deleted {
+			columnDict[column.Name] = columnID
+		}
+	}
+
+	schema := &TableSchema{
+		Table:      table,
+		ColumnDict: columnDict,
+	}
+
+	cf.Lock()
+	cf.schemas[table.Name] = schema
+	if _, tableExist := cf.enumMappings[table.Name]; !tableExist {
+		cf.enumMappings[table.Name] = make(map[int]enumDict)
+		cf.enumDefaultValueMappings[table.Name] = make(map[int]int)
+	}
+	for columnID, column := range table.Columns {
+		if !column.Deleted && column.IsEnumColumn() {
+			if _, columnExist := cf.enumMappings[table.Name][columnID]; !columnExist {
+				cf.enumMappings[table.Name][columnID] = make(enumDict)
+			}
+		}
+	}
+	cf.Unlock()
+	return schema
+}
+
+func (hf *httpSchemaFetcher) FetchAllEnums(tableName, columnName string) ([]string, error) {
+	var enumDictReponse []string
+
+	resp, err := hf.httpClient.Get(hf.enumDictPath(tableName, columnName))
+
+	err = hf.readJSONResponse(resp, err, &enumDictReponse)
+
+	return enumDictReponse, err
+}
+
+func (hf *httpSchemaFetcher) ExtendEnumCases(tableName, columnName string, enumCases []string) ([]int, error) {
+	enumCasesRequest := enumCasesWrapper{
+		EnumCases: enumCases,
+	}
+
+	enumCasesBytes, err := json.Marshal(enumCasesRequest)
+	if err != nil {
+		return nil, utils.StackError(err, "Failed to marshal enum cases")
+	}
+
+	var enumIDs []int
+	resp, err := hf.httpClient.Post(hf.enumDictPath(tableName, columnName), applicationJSONHeader, bytes.NewReader(enumCasesBytes))
+	err = hf.readJSONResponse(resp, err, &enumIDs)
+	if err != nil {
+		return nil, err
+	}
+	return enumIDs, nil
+}
+
+func (hf *httpSchemaFetcher) FetchSchema(tableName string) (*metaCom.Table, error) {
+	var table metaCom.Table
+	resp, err := hf.httpClient.Get(hf.tablePath(tableName))
+	err = hf.readJSONResponse(resp, err, &table)
+	if err != nil {
+		return nil, err
+	}
+
+	return &table, nil
+}
+
+func (hf *httpSchemaFetcher) FetchAllSchemas() ([]*metaCom.Table, error) {
```

Comment:
possible to combine into one http call?

Reply:
same here. maybe later when we see necessities for this change